File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/presenter/PresenterMainImpl.java
package pisarev.com.modeling.mvp.presenter;

import pisarev.com.modeling.interfaces.MainMvp;
import pisarev.com.modeling.mvp.model.MyFile;

public class PresenterMainImpl implements MainMvp.PresenterMainMvp {

    private final MainMvp.ViewMvp viewMvp;
    private final MyFile myFile;

    public PresenterMainImpl(MainMvp.ViewMvp viewMvp) {
        this.viewMvp = viewMvp;
        myFile = new MyFile();
    }

    @Override
    public void openProgram(String path) {
        viewMvp.showProgram(myFile.readFile(path));
    }

    @Override
    public void saveAll(String program, String path) {
    }

}

File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/presenter/PresenterMainImpl.kt
package pisarev.com.modeling.mvp.presenter

import pisarev.com.modeling.interfaces.MainMvp.ViewMvp
import pisarev.com.modeling.interfaces.MainMvp.PresenterMainMvp
import pisarev.com.modeling.mvp.model.MyFile

class PresenterMainImpl(private val viewMvp: ViewMvp) : PresenterMainMvp {
    private val myFile: MyFile
    override fun openProgram(path: String?) {
        viewMvp.showProgram(myFile.readFile(path))
    }

    override fun saveAll(program: String?, path: String?) {}

    init {
        myFile = MyFile()
    }
}